tests/NRedisStack.Tests/Person.cs

```diff
@@ -4,6 +4,6 @@ public class Person
     {
         public string? Name { get; set; }
         public int Age { get; set; }
-        public DateTime? Birthday;
+        public DateTime? Birthday { get; set; }
```

Comment:
Hi @shacharPash ,
I made Birthday a field other than a property on purpose, so it won't be serialized and deserialized by default. That's the reason the JsonSerializerOptions parameter is useful in Set and Merge method if developers want more control on serialization, in the test case, want the Birthday field to be serialized.